### ISSUE TITLE
GooglePaymentCardNonce does not return billing address

### ISSUE THREAD

**Issue by GrzegorzFeather** [braintree/braintree_android#169]
### General information

* SDK/Library version: 2.7.0
* Environment: Production
* Android Version and Device: Nexus 5X -> 8.0

### Issue description
When requesting Google Payments nonce, if I require billing address, the Nonce returned does not contain the billing address information:

Requesting address like this:
```
GooglePayment.requestPayment(btFragment, new GooglePaymentRequest()
              .transactionInfo(transactionInfo)
              .billingAddressRequired(true)
              .phoneNumberRequired(true));
```
Produces the following nonce with no billing address info:
<img width="792" alt="screen shot 2017-11-01 at 2 25 18 pm" src="https://user-images.githubusercontent.com/3044632/32298855-b853aa22-bf10-11e7-9d66-09c3941dfd31.png">

**Comment by lkorth** [comment]
The billing address is being returned from the Google Payments API, it just doesn't appear to contain all of the information you want. Have you tried [setting a format](https://github.com/braintree/braintree_android/blob/ad132ce10641b125bc1afc82d529a56b1788ce49/Braintree/src/main/java/com/braintreepayments/api/models/GooglePaymentRequest.java#L78)? 

The [Google docs on billing address format](https://developers.google.com/android/reference/com/google/android/gms/wallet/CardRequirements.Builder.html#setBillingAddressFormat(int)) include the following:

>Sets billing address format. See WalletConstants.BillingAddressFormat for more info. If omitted, defaults to BILLING_ADDRESS_FORMAT_MIN.
>
>Note that you should only set the format to BILLING_ADDRESS_FORMAT_FULL when it's required to process the order since it can increase friction during the checkout process and can lead to a lower conversion rate.

`BILLING_ADDRESS_FORMAT_MIN` looks to be exactly what you're getting back:

>When this format is used, the billing address returned will only contain the minimal info, including name, country code, and postal code.
>
>Note that some countries do not use postal codes, so the postal code field will be empty in those countries.

**Comment by GrzegorzFeather** [comment]
Hi @lkorth , so that did make it work for the address, but the phone number is still not there. 

Is there any other flag/value that I am missing?

```
GooglePayment.requestPayment(btFragment, new GooglePaymentRequest()
              .transactionInfo(transactionInfo)
              .billingAddressRequired(true)
              .billingAddressFormat(WalletConstants.BILLING_ADDRESS_FORMAT_FULL)
              .phoneNumberRequired(true));
```

<img width="322" alt="screen shot 2017-11-02 at 3 47 04 pm" src="https://user-images.githubusercontent.com/3044632/32354042-2faee206-bfe5-11e7-8853-0ab48aabcb97.png">

**Comment by lkorth** [comment]
I tested when requiring a shipping address in addition to the billing address just in case since the docs contain:

>Sets whether a phone number is additionally required from the buyer in both shipping and billing addresses (the phone number will only be returned if an address is required, otherwise this is a no-op). If omitted, defaults to false.

But I also did not receive one. It looks like it may be an issue in the Google Payments API, or the user does not have a phone number entered.

**Comment by lkorth** [comment]
This was fixed in 2.7.2.

**Comment by yarodevuci** [comment]
where to accuire this billing address? I have the same issue @lkorth
